### PR TITLE
Simplify signature of FieldMapper#parseCreateField.

### PR DESCRIPTION
**Pull request by jtibshirani**
`FieldMapper#parseCreateField` accepts the parse context, plus a list of fields
as an output parameter. These fields are immediately added to the document
through `ParseContext#doc()`.

This commit simplifies the signature by removing the list of fields, and having
the mappers add the fields directly to `ParseContext#doc()`. I think this is
nicer for implementors, because previously fields could be added either through
the list, or the context (through `add`, `addWithKey`, etc.)

### CODE REVIEW

**Comment by jpountz** [comment]
+1

**Review by cbuescher — APPROVED** [review]
LGTM, I left one suggestion, feel free to ignore.

File: modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/ScaledFloatFieldMapper.java [hunk 1]
```diff
@@ -442,9 +440,13 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
         boolean indexed = fieldType().indexOptions() != IndexOptions.NONE;
         boolean docValued = fieldType().hasDocValues();
         boolean stored = fieldType().stored();
-        fields.addAll(NumberFieldMapper.NumberType.LONG.createFields(fieldType().name(), scaledValue, indexed, docValued, stored));
+        List<Field> fields = NumberFieldMapper.NumberType.LONG.createFields(fieldType().name(), scaledValue, indexed, docValued, stored);
+        for (Field field : fields) {
```
[the remark this answers is not in the document]
Comment (jtibshirani) [reply modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/ScaledFloatFieldMapper.java]: 👍I updated to use the new method

**Comment by jtibshirani** [comment]
I pushed 77ac5d805bb37170c377b0ea5897f94e5d0e20a8 to switch over to more instances of `ParseContext.Document#addAll`.

**Pull request merged**